crates/sui-core/src/authority/authority_store_types.rs

```diff
@@ -77,7 +81,7 @@ impl From<Object> for StoreObjectPair {
             Data::Package(package) => StoreData::Package(package),
             Data::Move(move_obj) => {
                 // TODO: add real heuristic to decide if object needs to be stored indirectly
-                if cfg!(test) {
+                if cfg!(test) || cfg!(msim) {
```

Comment:
i don't think we should run tests in such a drastically different configuration. if we want to be able to exercise this in tests lets expose it via config - we'll want to do that anyway so that we can test indirect objects on one validator but not the other.